### PR TITLE
Reset Eventmanager upon runTx 

### PR DESCRIPTION
**Pull request by BrandonWeng**
## Describe your changes and provide context
Overwrite the event manager after each runTx since the events are only used in each runTx for parallelTx validation 

## Testing performed to validate your change
Unit tests

### CODE REVIEW

File: baseapp/test_helpers.go [hunk 1]
```diff
@@ -35,6 +41,10 @@ func (app *BaseApp) Deliver(txEncoder sdk.TxEncoder, tx sdk.Tx) (sdk.GasInfo, *s
 	}
 	ctx := app.deliverState.ctx.WithTxBytes(bz).WithVoteInfos(app.voteInfos).WithConsensusParams(app.GetConsensusParams(app.deliverState.ctx))
 	gasInfo, result, _, _, err := app.runTx(ctx, runTxModeDeliver, bz)
+
+	if len(ctx.MultiStore().GetEvents()) > 0 {
+		panic("Expected deliverTx events to be empty")
+	}
```
Comment (BrandonWeng) [inline baseapp/test_helpers.go:47]: Before the reset change this would fail

File: baseapp/test_helpers.go [hunk 2]
```diff
@@ -17,13 +17,19 @@ func (app *BaseApp) Check(txEncoder sdk.TxEncoder, tx sdk.Tx) (sdk.GasInfo, *sdk
 	}
 	ctx := app.checkState.ctx.WithTxBytes(bz).WithVoteInfos(app.voteInfos).WithConsensusParams(app.GetConsensusParams(app.checkState.ctx))
 	gasInfo, result, _, _, err := app.runTx(ctx, runTxModeCheck, bz)
+	if len(ctx.MultiStore().GetEvents()) > 0 {
+		panic("Expected checkTx events to be empty")
```
Comment (philipsu522) [inline baseapp/test_helpers.go:21]: does it make sense to panic if its just events? this isn't on the critical path so it seems strange to error out b/c of this. Perhaps a warning log instead?

Comment (BrandonWeng) [reply baseapp/test_helpers.go:21]: yeah, this is just unit test

**Review by yzang2019 — APPROVED** [review]
Verified in a previously OOM node and confirmed the issue got fixed, thanks!

File: baseapp/test_helpers.go (hunk 2 above)
Comment (philipsu522) [reply baseapp/test_helpers.go:21]: oo ops

**Review by philipsu522 — APPROVED** [review]

**Pull request merged**